app.js

```diff
@@ -786,12 +786,26 @@ if (opts.version) {
     logger.info(`  git release ${gitReleaseName}`);
     logger.info(`  release build ${releaseBuildNumber}`);
     logger.info('Exiting');
-    process.exit(0);
+    terminationHandler('', 0)();
+}
+
+process.on('uncaughtException', terminationHandler('uncaughtException', 1));
+process.on('SIGINT', terminationHandler('SIGINT', 0));
+process.on('SIGTERM', terminationHandler('SIGTERM', 0));
+process.on('SIGQUIT', terminationHandler('SIGQUIT', 0));
+
+function terminationHandler(name, code) {
+  return (error) => {
+    logger.info(`stopping process: ${name}`);
+    if (error && error instanceof Error) {
+      logger.error(error);
+    }
+    process.exit(code);
+  };
 }
 
 main()
-    .then(() => null)
     .catch(err => {
         logger.error('Top-level error (shutting down):', err);
-        process.exit(1);
+        terminationHandler('Top-level error', 0)(err);
```

Comment:
As noted, would prefer this remain `process.exit`, but if this line does stay the exit code must remain 1

Reply:
I can remove that, to be like this:

```suggestion
        process.exit(1);
```

Reply:
Fixed! Thanks for reviewing!